### DISCUSSION TITLE
Alternative approach for GameImages

### DISCUSSION THREAD

**Discussion by malnaanah** [ruanformigoni/gameimage#117] [Ideas]
I love the idea behind gameImages and how I can run a windows game in Linux from a single file; however, it consumes large amount of memory and storage. I am thinking of a different approach and I wonder if it exists or can be done.
1- Having separate appimage for wine that is used to run other flatimages (or appimages) that contains Windows and game files.
2- The separate flatimages or appimage can be read-only and the read+write folders like "Users" folder can be linked into the user's home folder.
3- No wrapping is needed and memory usage should be minimized.

Does this idea exist or can it be done?

**Comment by ruanformigoni** [comment]
Hi @malnaanah , thanks for the suggestions.

1. I have used AppImages in this project before, and they proved to not be portable. If you build one in archlinux, you cannot use it in debian nor mint, since it has different glibc versions. Also there is no containerization, you have to do it yourself, this makes wine have several issues by accessing libraries from the host erroneously. It was an overall a bad experience that I will not go back to, and the reason why flatimage exists.
2. That is possible to do in flatimage already, the default is to not touch the user default home directory and have all the game files in the `.game.flatimage.config` directory. To change that, and make for example the `Documents` folder from wine to point to the `Documents` folder in your home directory, you can use `./game.flatimage fim-bind add rw '/home/gameimage/Documents' '/home/malnaanah/Documents'.
3. If you reduce the memory usage, by disabling bubblewrap, it is not portable to other linux distros anymore. Is this memory overhead also present is lighter version of wine that do not use UMU, such as caffe? Maybe that will lighten the toll in your system.
